flang/lib/Optimizer/CodeGen/TypeConverter.h

```diff
@@ -55,8 +55,11 @@ class LLVMTypeConverter : public mlir::LLVMTypeConverter {
       LLVM_DEBUG(llvm::dbgs() << "type convert: " << boxchar << '\n');
       return convertType(specifics->boxcharMemoryType(boxchar.getEleTy()));
     });
-    addConversion(
-        [&](BoxProcType boxproc) { return convertBoxProcType(boxproc); });
+    addConversion([&](BoxProcType boxproc) {
+      // TODO: Support for this type will be added later when the Fortran 2003
+      // procedure pointer feature is implemented.
+      return llvm::None;
+    });
```

Comment:
Shouldn't this be a TODO/crash? What happens here?

Reply:
This returns a type conversion failure. 
For fir-dev I have retained the original code. If required I can change the upstream code to a hard TODO.